### ISSUE TITLE
Unable to deploy contracts using truffle v5.0.16 with contract solidity version 0.5.0

### ISSUE THREAD

**Issue by carnelius8888** [trufflesuite/truffle#2012]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

What the issue is, in broad strokes.

I am able to compile the contracts but unable to deploy them on my current Quorum Setup.

[CheckingContract.txt](https://github.com/trufflesuite/truffle/files/3171635/CheckingContract.txt)
[Dispatcher.txt](https://github.com/trufflesuite/truffle/files/3171636/Dispatcher.txt)
[Dispatcher2.txt](https://github.com/trufflesuite/truffle/files/3171637/Dispatcher2.txt)
[DispatcherStorage.txt](https://github.com/trufflesuite/truffle/files/3171638/DispatcherStorage.txt)
[DispatcherStorgae2.txt](https://github.com/trufflesuite/truffle/files/3171639/DispatcherStorgae2.txt)
[Lib1.txt](https://github.com/trufflesuite/truffle/files/3171640/Lib1.txt)
[Lib2.txt](https://github.com/trufflesuite/truffle/files/3171641/Lib2.txt)
[Migrations.txt](https://github.com/trufflesuite/truffle/files/3171642/Migrations.txt)
[Lib1Storage.txt](https://github.com/trufflesuite/truffle/files/3171646/Lib1Storage.txt)

## Steps to Reproduce

Please provide the shortest amount of steps to reproduce your issue.
Please try to deploy it on Quorum. It compiles it without error but never deploys it and gives contract address.
## Expected Behavior
It will only compile these contract but will not deploy them at truffle migrate command
![OnlyCompile-No-Deployment](https://user-images.githubusercontent.com/26059780/57604220-4166d500-7581-11e9-86c8-34db5cc0fae1.png)

What you expected to happen.
Giving contract addresses after deployment
## Actual Results

What actually happened. Please give examples and support it with screenshots, copied output or error messages.

## Environment

* Operating System: 18.04
* Ethereum client:
* Truffle version (`truffle version`):Truffle v5.0.16
* node version (`node --version`):v8.10.0
* npm version (`npm --version`): 3.5.2

**Comment by carnelius8888** [comment]
const config = require("./config.json");
var ipa = config.node1;
module.exports = {
    networks: {
        development: {
            host: ipa.host,
            port: ipa.port,
            network_id: "*", // Match any network id
            gasPrice: 0,
            gas: 1000000
        }
    },
    compilers: {
        solc: {
            version: "0.5.1" // Fetch exact version from solc-bin (default: truffle's version)
             // docker: true,        // Use "0.5.1" you've installed locally with docker (default: false)
        }
    }
};

**Comment by carnelius8888** [comment]
![Error_Screenshot](https://user-images.githubusercontent.com/26059780/57606435-7e819600-7586-11e9-9275-ff05e5ef7151.png)
Adding new Error Screenshot.

**Comment by CruzMolina** [comment]
Hey @carnelius8888! It looks like you haven't configured `truffle.js` to use a Quorum network type.

From the [docs](https://truffleframework.com/docs/truffle/getting-started/working-with-quorum#configuration):
>To use Quorum, you must modify your network in truffle-config.js to include a parameter type set to "quorum". See the example below.

```module.exports = {
  networks: {
    development: {
      host: "127.0.0.1",
      port: 22000, // replace with quorum node port you wish to connect to
      network_id: "*",
      type: "quorum"
    }
  }
};```

Hope this helps! 🤞

**Comment by yaoshiang** [comment]
This may be a bug I just ran into: 

truffle deploy is supposed to be an alias for truffle migrate. At least in 5.0.17, it is not. I got this error message. 

```
truffle deploy

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.

TypeError: this.determineDryRunSettings is not a function
    at Contracts.compile.then (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/commands/migrate.js:184:10)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
Truffle v5.0.17 (core: 5.0.16)
Node v8.10.0

```

Solution? 

run

`truffle migrate`

**Comment by naddison36** [comment]
Thanks @yaoshiang. I was struggling with the same error as I was running `truffle deploy`. `truffle migrate` works.

**Comment by eggplantzzz** [comment]
Yes, this bug is fixed in PR #2018

**Comment by eggplantzzz** [comment]
Closing this presuming it is no longer a problem. If I'm wrong, feel free to open a new one!
